README.md

```diff
@@ -0,0 +1,494 @@
+    * `meta` [`<Object>`][] - Per-message meta data.  Combined with key-values created with `addMetaProperty`
+* Emits: [warn](#event-warn), [error](#event-error)
+
+Sends a string or object to LogDNA for storage.  If the [convenience methods](#convenience-methods) are used, they call this function
+under the hood, so the options are the same.  The only difference is that `level` is automatically set in the convenience methods.
+
+
+### `logger.removeMetaProperty(key)`
+
+* `key` [`<String>`][] - The key (and implied value) to be remove from the global `meta` object.
+* Emits: [`warn`](#event-warn)
+
+Attempts to remove the given key from the global `meta` data object.  If the key is not found, `warn` is emitted.
+
+
+## How Log Lines are Sent to LogDNA
+
+In default operation, when `log` functions are called, the line is added to a buffer to cut down on HTTP traffic to the server.  
+The buffer is flushed every `flushIntervalMs` milliseconds, or if the log line lengths grow beyond `flushLimit`.
+
+When `flush` fires (or is called manually), the current buffer is put onto a send queue, and a new buffer is started.  The send queue begins
+sending to LogDNA.  It will continue to send without pausing or honoring `flushIntervalMs` as long as there are buffers in the send queue.  
+When the send queue is empty, `cleared` is emitted.
+
+### Handling Errors
+
+* `timeout` or `500` errors will be retried using an [exponential backoff strategy](#exponential-backoff-strategy), but will
+  also emit `error` events along the way.
+* User-level errors (such as `400`) will not be retried because they most likely would never be successful (if the message is deemed invalid),
+  and `error` events are emitted for these also.
+
+
+## Exponential Backoff Strategy
+
+When HTTP failures happen, if they are deemed "retryable" (non-user errors), then the client will pause for a short time before
+trying to resend.  The algorithm it implements is an exponential backoff with "Jitter" strategy that uses random numbers to statistically
+spread out the wait times in order to avoid flooding.
+
+The settings for `baseBackoffMs` and `maxBackoffMs` are used in this algorithm, and serve as the lower and upper boundaries for the wait time.
+
+These types of errors are blocking since they are related to timeouts and server errors.  Logs will continue to buffer as normal, and
+if the HTTP calls becomes successful, they will begin to send immediately, and without pause.
+
+
+## Best Practices
+
+* The client is optimized for high throughput.  Using a single instance is no problem, but multiple can be created with the same key if desired.
+* Set up an `error` listener so that your app is aware of problems.  Things like HTTP errors are emitted this way.
+* When shutting down your application, make sure all log entries are cleared.  Services like AWS Lambda can buffer log entries,
+  so it might be worthwhile to pause for a short time before exiting:
```

Comment:
```suggestion
* When shutting down your application, ensure all log entries are cleared. Services like AWS Lambda can buffer log entries,
  so it might be worthwhile to pause for a short time before exiting like in the following example:
```